### PR TITLE
Block public access to S3 bucket and its objects by default

### PR DESCRIPTION
**Pull request by coreydaley**
Blocks public access to the S3 bucket that we create and the objects that it contains along with tests to verify
https://docs.aws.amazon.com/AmazonS3/latest/dev/access-control-block-public-access.html

### CODE REVIEW

**Comment by dmage** [comment]
To be consistent, we should do the same actions during upgrade for managed buckets that were created by 4.1.

File: pkg/apis/imageregistry/v1/types.go [hunk 1]
```diff
@@ -59,6 +59,10 @@ const (
 	// that we created has encryption enabled
 	StorageEncrypted = "StorageEncrypted"
 
+	// StoragePublicAccessBlocked denotes where or not the registry storage medium
```
Comment (bparees) [inline pkg/apis/imageregistry/v1/types.go]: s/where/whether/

**Comment by bparees** [comment]
From the doc you linked:
```
By default, new buckets and objects don't allow public access, but users can modify bucket policies or object permissions to allow public access. 
```

so why is this needed?

**Comment by coreydaley** [comment]
I believe that it keeps the bucket/objects from being exposed publicly by accident due to any other access controls being set/modified.

> From the doc you linked:
> 
> ```
> By default, new buckets and objects don't allow public access, but users can modify bucket policies or object permissions to allow public access. 
> ```
> 
> so why is this needed?

File: pkg/apis/imageregistry/v1/types.go (hunk 1 above)
Comment (coreydaley) [reply pkg/apis/imageregistry/v1/types.go]: updated in next push

**Comment by coreydaley** [comment]
> To be consistent, we should do the same actions during upgrade for managed buckets that were created by 4.1.

Are you ok with doing that in a follow up pr?  I think that it requires a little bit more re-working of what should and should not be run when the sync occurs and checks that should be made first vs what can just be re-applied without causing an error.  Along with thoughts on whether or not we should change settings back to what we want them to be if the admin has changed x setting on the s3 storage themselves.

**Comment by coreydaley** [comment]
@dmage pushed updates, ptal

Fixed grammar issue identified by @bparees and nil pointer issue.

File: pkg/storage/s3/s3.go [hunk 2]
```diff
@@ -328,6 +328,31 @@ func (d *driver) CreateStorage(cr *imageregistryv1.Config) error {
 		return err
 	}
 
+	// Block public access to the s3 bucket and it's objects by default
```
Comment (dmage) [inline pkg/storage/s3/s3.go]: `it's`?

Comment (coreydaley) [reply pkg/storage/s3/s3.go]: Fixed

**Comment by coreydaley** [comment]
@dmage fixed grammar, ptal

**Pull request merged**